framework/src/base/MooseServer.C

```diff
@@ -0,0 +1,529 @@
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  // TODO - hook up this capability by adding server specific logic here
+
+  is_incomplete = pass;
+
+  return pass;
+}
+
+bool
+MooseServer::gatherDocumentDefinitionLocations(wasp::DataArray & definitionLocations,
+                                               int line,
+                                               int character)
+{
+  (void)definitionLocations; // suppress unused variable warning
+  (void)line;                // suppress unused variable warning
+  (void)character;           // suppress unused variable warning
+
+  if (!this->is_setup)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be setup" << std::endl;
+    return false;
+  }
+
+  if (!this->is_initialized)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be initialized" << std::endl;
+    return false;
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  // TODO - hook up this capability by adding server specific logic here
+
+  return pass;
+}
+
+bool
+MooseServer::gatherDocumentReferencesLocations(wasp::DataArray & referencesLocations,
+                                               int line,
+                                               int character,
+                                               bool include_declaration)
+{
+  (void)referencesLocations; // suppress unused variable warning
+  (void)line;                // suppress unused variable warning
+  (void)character;           // suppress unused variable warning
+  (void)include_declaration; // suppress unused variable warning
+
+  if (!this->is_setup)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be setup" << std::endl;
+    return false;
+  }
+
+  if (!this->is_initialized)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be initialized" << std::endl;
+    return false;
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  // TODO - hook up this capability by adding server specific logic here
+
+  return pass;
+}
+
+bool
+MooseServer::gatherDocumentFormattingTextEdits(wasp::DataArray & formattingTextEdits,
+                                               int start_line,
+                                               int start_character,
+                                               int end_line,
+                                               int end_character,
+                                               int tab_size,
+                                               bool insert_spaces)
+{
+  (void)formattingTextEdits; // suppress unused variable warning
```

Comment:
see above

Reply:
Good feedback - I just added a commit addressing this.